Explorer/Assets/DCL/SDKComponents/Tween/Systems/TweenUpdaterSystem.cs

```diff
@@ -113,138 +117,99 @@ private void HandleComponentRemoval(ref SDKTweenComponent tweenComponent, ref CR
         }
 
-        private Tweener SetupPositionTween(Transform entityTransform, Vector3 startPosition,
-            Vector3 endPosition, float durationInSeconds, Ease ease, bool faceDirection)
+        private void UpdateTweenPosition(CRDTEntity sdkEntity, SDKTweenComponent sdkTweenComponent, ref SDKTransform sdkTransform)
         {
-            if (faceDirection) entityTransform.forward = (endPosition - startPosition).normalized;
-
-            entityTransform.localPosition = startPosition;
-            return entityTransform.DOLocalMove(endPosition, durationInSeconds).SetEase(ease).SetAutoKill(false);
+            TweenSDKComponentHelper.WriteTweenResult(ref sdkTransform, sdkTweenComponent.CustomTweener);
+            TweenSDKComponentHelper.WriteTweenResultInCRDT(ecsToCRDTWriter, sdkEntity, sdkTweenComponent.CustomTweener);
         }
 
-        private TweenStateStatus GetCurrentTweenState(float currentTime, bool isPlaying)
+
+        private void SetupTweener(CRDTEntity entity, ref SDKTweenComponent sdkTweenComponent, Transform entityTransform, PBTween tweenModel, float durationInSeconds, bool isPlaying, SDKTransform sdkTransform)
         {
-            if (!isPlaying) { return TweenStateStatus.TsPaused; }
+            //NOTE: Left this per legacy reasons, Im not sure if this can happen in new renderer
+            // There may be a tween running for the entity transform, e.g: during preview mode hot-reload.
+            if (sdkTweenComponent.IsActive())
+            {
+                sdkTweenComponent.Rewind();
+                TweenSDKComponentHelper.WriteTweenResult(ref sdkTransform, sdkTweenComponent.CustomTweener);
+                TweenSDKComponentHelper.WriteTweenResultInCRDT(ecsToCRDTWriter, entity, sdkTweenComponent.CustomTweener);
+                tweenerPool.Return(sdkTweenComponent);
+            }
+
+            if (!EASING_FUNCTIONS_MAP.TryGetValue(tweenModel.EasingFunction, out Ease ease))
+                ease = Linear;
 
-            return currentTime.Equals(1f) ? TweenStateStatus.TsCompleted : TweenStateStatus.TsActive;
+            sdkTweenComponent.CustomTweener = tweenerPool.GetTweener(tweenModel, entityTransform, durationInSeconds);
+            sdkTweenComponent.CustomTweener.ParentId = sdkTransform.ParentId;
+            sdkTweenComponent.CustomTweener.DoTween(ease, tweenModel.CurrentTime * durationInSeconds, isPlaying);
         }
 
-        private Tweener SetupRotationTween(Transform entityTransform, Quaternion startRotation,
-            Quaternion endRotation, float durationInSeconds, Ease ease)
+        private void CleanUpTweenBeforeRemoval(CRDTEntity sdkEntity, SDKTweenComponent sdkTweenComponent)
         {
-            entityTransform.localRotation = startRotation;
-            return entityTransform.DOLocalRotateQuaternion(endRotation, durationInSeconds).SetEase(ease).SetAutoKill(false);
+            tweenerPool.Return(sdkTweenComponent);
```

Comment:
I propose to nullify `CustomTweener` on returning to the pool as there is a bit of logic and it might gone terribly wrong if the instance returned to the pool is preserved in the component. It would additional protection from it as we already faced similar bugs in the past.